doc/specification.md

```diff
@@ -86,15 +86,13 @@ A client expresses modifications to the RIB modification by sending a set of `AF
 
 `AFTOperation` is identified by its `id`.  The `AFTOperation.id` should be unique per `Modify` RPC session. It's the client's responsibility to gurrantee the uniqueness during a `Modify` RPC session.
 
-#### 4.1.3.1 AFTOperation Validation
-* minimum validation.
-* `election_id` check.
+#### 4.1.3.1 AFTOperation Content Validation
 
-#### 4.1.3.2 Forward References
+gRIBI relies strictly on the clients being conservative in what they send. It is the clients' responsiblity to ensure the correctness of AFTOperations content. However, the receiving device SHOULD determine each entry's viability before installing it to RIB/FIB (i.e., is the next-hop set resolvable).
```

Comment:
I think it might be worth saying here that the gRIBI daemon expects that the client has built a well-formed request, and does not perform validation before sending it to the RIB. I expect that the device probably does some form of validation at the RIB - right?

I think the forward reference case you have below is actually counter to this principle -- we probably need to say what it is the gRIBI server should do:

 * ensure that the message is syntactically valid
 * check forward references

other than this, it will not check any other parameters, and will rather rely on the RIB to reject invalid contents. Whether an entry becomes used is dependent upon the RIB and not gRIBI -- i.e., it should not be expected that the device checks that the ACK mode was FIB_ACK and determines that the new entry will be eligible to programmed into the FIB.

Reply:
let's discuss it offline first.

Reply:
ACK - let's sync up when we talk next.

Reply:
Updated the wording per our discussion.